test/export/test_torchbind.py

```diff
@@ -1,36 +1,83 @@
 # Owner(s): ["oncall: export"]
-import unittest
 
 import torch
+import torch.testing._internal.torchbind_impls  # noqa: F401
+import torch.utils._pytree as pytree
+from torch._functorch.aot_autograd import aot_export_module
+from torch._higher_order_ops.effects import _EffectType, SIDE_EFFECTS
 from torch._higher_order_ops.torchbind import enable_torchbind_tracing
 from torch.export import export
-from torch.testing._internal.common_utils import (
-    find_library_location,
-    IS_FBCODE,
-    IS_MACOS,
-    IS_SANDCASTLE,
-    IS_WINDOWS,
-    run_tests,
-    skipIfTorchDynamo,
-    TestCase,
-)
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo, TestCase
 
 
 @skipIfTorchDynamo("torchbind not supported with dynamo yet")
 class TestExportTorchbind(TestCase):
     def setUp(self):
-        if IS_MACOS:
-            raise unittest.SkipTest("non-portable load_library call used in test")
-        elif IS_SANDCASTLE or IS_FBCODE:
-            torch.ops.load_library(
-                "//caffe2/test/cpp/jit:test_custom_class_registrations"
-            )
-        elif IS_WINDOWS:
-            lib_file_path = find_library_location("torchbind_test.dll")
-            torch.ops.load_library(str(lib_file_path))
-        else:
-            lib_file_path = find_library_location("libtorchbind_test.so")
-            torch.ops.load_library(str(lib_file_path))
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_Foo")
+        class FakeFoo:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+            @classmethod
+            def from_real(cls, foo):
+                (x, y), _ = foo.__getstate__()
+                return cls(x, y)
+
+            def add_tensor(self, z):
+                return (self.x + self.y) * z
+
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_TensorQueue")
+        class FakeTensorQueue:
+            def __init__(self, q):
+                self.queue = q
+
+            @classmethod
+            def from_real(cls, real_tq):
+                return cls(real_tq.clone_queue())
```

Comment:
1. clone_queue() is not good enough because the strides on the cloned queue may be different from the strides on the original queue. The way to go about it is to use FakeTensorConverter to convert each element of the queue.

Reply:
We should probably add an explicit test for this (that takes a TensorQueue and creates a FakeTensorQueue and then checks the properties of the Tensors)

Reply:
Yeah, but this from_real is written by users. We won't want them to use FakeTensorConverter directly... Let me figure out how to deal with it. Right now, the tensors inside the object is not fakified yet.

Reply:
Does directly using the FakeTensor constructor hit FakeTensorConverter?

Reply:
no. FakeTensorMode.from_real hits the converter.